src/vanilla.ts

```diff
@@ -1,7 +1,7 @@
 export type State = Record<string | number | symbol, unknown>
-export type PartialState<T extends State> =
-  | Partial<T>
-  | ((state: T) => Partial<T>)
+export type PartialState<T extends State, K extends keyof T = keyof T> =
+  | (Pick<T, K> | T)
+  | ((state: T) => Pick<T, K> | T)
```

Comment:
Do you know if we really need `| T`?

Reply:
I do not :)

I got this from the react typings on `setState` of a class component:

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/6c49e45842358ba59a508e13130791989911430d/types/react/v16/index.d.ts#L489-L495

Judging from the comment, it is needed for intelliSense 😅

Reply:
Ah ha. We need to put the link in the code comment, then.

Reply:
done here: b142dcda